OnlineShop/Controllers/ItemsController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using DataAccess.EF;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using DataAccess.EF.Models;
+
+namespace OnlineShop.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ItemsController : ControllerBase
+    {
+        public ItemsController(IUnitOfWork uow)
+        {
+            _uow = uow;
+        }
+
+        [HttpGet]
+        public ActionResult<IEnumerable<Item>> GetItems([FromQuery] int categoryId, [FromQuery] int page, [FromQuery] int limit)
+        {
+            return _uow.ItemRepository
+                .Get(i => i.CategoryId.Equals(categoryId))
+                .Skip((page - 1) * limit)
+                .Take(limit)
+                .ToList();
+        }
+
+        [HttpGet("{id:int}")]
+        public ActionResult<Item?> GetItem(int id)
+        {
+            return _uow.ItemRepository.Get(i => i.Id.Equals(id)).SingleOrDefault();
+        }
+
+        [HttpPut("{id:int}")]
+        public IActionResult PutItem(int id, Item? item)
+        {
+            if (item == null || id != item.Id) return BadRequest();
```

Comment:
the same as for categories. Also looks like both of the are some uniformed entities, so interface with id, and filter would be better solution in case you have to imlement update in this way